Aplib.Core/LiftingExtensionMethods.cs

```diff
@@ -0,0 +1,121 @@
+using Aplib.Core.Belief.BeliefSets;
+using Aplib.Core.Desire.DesireSets;
+using Aplib.Core.Desire.Goals;
+using Aplib.Core.Desire.GoalStructures;
+using Aplib.Core.Intent.Actions;
+using Aplib.Core.Intent.Tactics;
+
+namespace Aplib.Core
+{
+    /// <summary>
+    /// Contains extension methods for lifting BDI cycle components into higher-order components.
+    /// </summary>
+    public static class LiftingExtensionMethods
+    {
+        /// <summary>
+        /// Wraps a normal action into a tactic.
+        /// </summary>
+        /// <returns>
+        /// A primitive tactic, whose guard always returns true.
+        /// If the action implements <see cref="IDocumented"/>, it's name will be prefixed with an indicator that it
+        /// has been lifted.
+        /// </returns>
+        /// <param name="action">
+        /// The action which on its own can function as a tactic. Meaning, the tactic consists of just a single action.
+        /// </param>
+        public static PrimitiveTactic<TBeliefSet> Lift<TBeliefSet>(this IAction<TBeliefSet> action)
+            where TBeliefSet : IBeliefSet
+        {
+            string mostDerivedActionType = action.GetType().Name;
+
+            IMetadata metadata = action is not IDocumented actionWithMetadata
+                ? new Metadata($"Lifted {mostDerivedActionType}")
+                : new Metadata(!string.IsNullOrWhiteSpace(actionWithMetadata.Metadata.Name)
+                    ? $"[Lifted {mostDerivedActionType}] {actionWithMetadata.Metadata.Name}"
+                    : $"Lifted {mostDerivedActionType}",
+                    actionWithMetadata.Metadata.Description);
```

Comment:
I always include the metadata, for its a feature and something someone should rely on: if something's lifted, you should know. To rely on that, I do not make it optional.

Besides, if I implicitly lift, I would still need to check if you properly implemented the metadata. It partially moves the problem.

If you look at lifting manually, I am not in favor of optionally leaving out the previous metadata. It resets all info previously known, kind of destructive. Want another metadata? The user can set it himself. Does not matter if he does that as an argument, or by setting it manually aftewards, same number of characters (roughly).

Reply:
1. That destructiveness of the metadata is the same behavior that you get when not using lifting. For example, let's say you have an action A, and want to create a primitive tactic from it. The code `new PrimitiveTactic(A)` will create a primitive tactic without metadata. It does not reset any data when lifting, because the underlying action still has its metadata after lifting with the lift method. No information is lost.
2. Cannot provide your own metadata when lifting. Sure, you can do it afterward, but why not give it as a parameter when you're at it? If we want to let the user do it afterward, then why not do so for _all_ other constructors as well?
3. If something's lifted, you _do_ know, because you lifted it.